### PR TITLE
Add testing.patch which calls mock.patch with wraps argument

### PR DESCRIPTION
**Pull request by niboshi**
Added `testing.patch` and fixed NaN issue for cuDNN tests (#3878)
Note that this PR does not fix all of #3878.

### CODE REVIEW

**Comment by niboshi** [comment]
Conflict resolved.

File: chainer/testing/helper.py [hunk 1]
```diff
@@ -50,3 +66,39 @@ def assert_warns(expected):
...
+        target(str): Full name of target object.
+        wraps: Wrapping object which will be passed to ``mock.patch`` as
+            ``wraps`` argument.
+            If omitted, the object specified by ``target`` is used.
+        *args: Passed to ``mock.patch``.
+        **kwargs: Passed to ``mock.patch``.
+
+    """
+    _check_mock_available()
+    wraps = kwargs.pop('wraps', _default)
```
Comment (okuta) [inline chainer/testing/helper.py]: How about using `try except` ?

**Comment by okuta** [comment]
LGTM except comment.

File: chainer/testing/helper.py (hunk 1 above)
Comment (niboshi) [reply chainer/testing/helper.py]: Thanks. Fixed.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**